interview-questions.md

```diff
@@ -10,32 +10,32 @@ Type your initials here to confirm you read the instructions:
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is the input function definiton while an argument is the actual value passed in to the function.
 
-Researched answer:
+Researched answer: In functions, there are parameters and arguments. Parameters act as placeholders for information, or arguments, to be passed in when a function is being used.
 
 2. What is the difference between an object and a class?
 
-Your answer:
+Your answer: A class is just the blueprint for an object.
 
-Researched answer:
+Researched answer: An object is a data type that holds collections of information in the form of pairs. A class is the blueprint for objects and, like functions, can be reused and customized as needed.
 
 3. What is iteration?
 
-Your answer:
+Your answer: Iteration is to execute a set of instructions over and over again until a condition is met.
 
-Researched answer:
+Researched answer: In Javascript, loops are needed when we need to execute code over items in a code, typically elements in an array, until a condition is met. The process of repeating a set of instructions over and over again until a condition is met is called iteration.
```

Comment:
Loops are a way to start iteration, but they are not required because there are many ways to start iteration